src/pages/VenuePage/VenuePage.tsx

```diff
@@ -128,7 +133,18 @@ export const VenuePage: React.FC<VenuePageProps> = ({
     });
   }, LOC_UPDATE_FREQ_MS);
 
-  const { sovereignVenueId, sovereignVenueDescendantIds } = useRelatedVenues();
+  const {
+    sovereignVenueId,
+    sovereignVenueDescendantIds,
+    parentVenue,
+  } = useRelatedVenues({ currentVenueId: space.id });
+
+  useEffect(() => {
+    setBackButtonTargetSpace(parentVenue);
+    return () => {
+      setBackButtonTargetSpace(undefined);
```

Comment:
[NOTE]
If only there were few simple standard hooks with extra utilities to avoid passing `undefined`. Looks like a code smell while `clearBackButtonTargetSpace()` would not.